build-system/tasks/helpers.js

```diff
@@ -385,6 +384,49 @@ function finishBundle(srcFilename, destDir, destFilename, options) {
   }
 }
 
+/**
+ * Creates a babel plugin for esbuild for the given caller. Optionally enables
+ * caching to speed up transforms.
+ * @param {string} callerName
+ * @param {boolean} enableCache
+ * @param {function()} postStep
+ * @return {!Object}
+ */
+function getEsbuildBabelPlugin(callerName, enableCache, postStep = () => {}) {
+  return {
+    name: 'babel',
+    async setup(build) {
+      const transformContents = async (file) => {
+        const contents = await fs.promises.readFile(file.path, 'utf-8');
+        const babelOptions = babel.loadOptions({
+          caller: {name: callerName},
+          filename: file.path,
+          sourceFileName: path.basename(file.path),
+        });
+        const result = await babel.transformAsync(contents, babelOptions);
+        return {contents: result.code};
+      };
+
+      build.onLoad({filter: /.*\.[cm]?js$/, namespace: ''}, async (file) => {
+        const {path} = file;
+        if (enableCache && cache.has(path)) {
+          return cache.get(path);
+        }
+        const promise = transformContents(file);
+        if (enableCache) {
+          // Cache needs to be set before awaiting, because esbuild can issue
+          // multiple "loads" to import a file while waiting for babel to
+          // transform.
+          cache.set(path, promise);
+        }
+        const transformed = await promise;
+        postStep();
```

Comment:
are there any usage of `postStep` ? wanted to understand uses cases

Reply:
I believe it is for outputting "......." while babel is transforming files

Reply:
Yep, dot progress.